kotlin/internal/jvm/compile.bzl

```diff
@@ -266,14 +266,12 @@ def _fold_jars_action(ctx, rule_kind, toolchains, output_jar, input_jars, action
 def _resourcejar_args_action(ctx):
     res_cmd = []
     for f in ctx.files.resources:
-        c_dir, res_path = _adjust_resources_path(f.short_path, ctx.attr.resource_strip_prefix)
-        target_path = res_path
+        target_path = _adjust_resources_path(f.short_path, ctx.attr.resource_strip_prefix)
         if target_path[0] == "/":
             target_path = target_path[1:]
-        line = "{target_path}={c_dir}{res_path}\n".format(
-            res_path = res_path,
+        line = "{target_path}={f_path}\n".format(
```

Comment:
Thanks for this change @jdai8! I was so confused why zipping was failing for some resource files generated under bazel-out.

To be super clear this argument line `{target_path}={c_dir}{res_path}` is saying that the file located at `{c_dir}{res_path}` should be zipped under `{target_path}` for the resource jar. So moving to `{target_path}={f_path}` should be strictly just as correct, because `f.path` should always be how input files are located by a bazel action and `{target_path}` remains the same.